crates/ika-core/src/dwallet_mpc/mpc_manager.rs

```diff
@@ -635,27 +317,142 @@ impl DWalletMPCManager {
             "Received start MPC flow event for session identifier {:?}",
             session_identifier
         );
+        let with_mpc_event_data = if mpc_event_data.is_some() {
+            true
+        } else {
+            false
+        };
 
         let new_session = DWalletMPCSession::new(
-            self.epoch_store.clone(),
-            self.consensus_adapter.clone(),
-            self.epoch_id,
+            self.validator_name,
             MPCSessionStatus::Active,
             *session_identifier,
             self.party_id,
-            self.weighted_threshold_access_structure.clone(),
+            self.access_structure.clone(),
             mpc_event_data,
-            self.dwallet_mpc_metrics.clone(),
-            self.root_seed.clone(),
+            self.network_dkg_third_round_delay,
+            self.decryption_key_reconfiguration_third_round_delay,
         );
+
         info!(
-            // todo(zeev): add metadata.
+            party_id=self.party_id,
+            authority=?self.validator_name,
+            with_mpc_event_data,
+            ?session_identifier,
             last_session_to_complete_in_current_epoch=?self.last_session_to_complete_in_current_epoch,
-            "Adding MPC session to active sessions",
+            "Adding a new MPC session to the active sessions map",
         );
+
         self.mpc_sessions.insert(*session_identifier, new_session);
     }
 
+    /// Spawns all ready MPC cryptographic computations on separate threads using Rayon.
+    /// If no local CPUs are available, computations will execute as CPUs are freed.
+    ///
+    /// A session must have its `mpc_event_data` set in order to be advanced.
+    ///
+    /// System sessions are always advanced if a CPU is free, user sessions are only advanced
+    /// if they come before the last session to complete in the current epoch (at the current time).
+    ///
+    /// System sessions are always advanced before any user session,
+    /// and both system and user sessions are ordered internally by their sequence numbers.
+    ///
+    /// The messages to advance with are built on the spot, assuming they satisfy required conditions.
+    /// They are put on a `ComputationRequest` and forwarded to the `orchestrator` for execution.
+    pub(crate) async fn perform_cryptographic_computation(
+        &mut self,
+    ) -> HashMap<
+        ComputationId,
+        DwalletMPCResult<
+            mpc::AsynchronousRoundResult<
+                MPCMessage,
+                MPCPrivateOutput,
+                SerializedWrappedMPCPublicOutput,
+            >,
+        >,
+    > {
+        let computation_requests: Vec<_> = self
```

Comment:
I would split it to vars with meaning full names, e.g. mpc_sessions_with_event_data, sorted_mpc_sessions_with_event_data and computation_requests